### ISSUE TITLE
Riot cannot start due to nonfinishing missing e2ee / cross-signing endpoints

### ISSUE THREAD

**Issue by grinapo** [matrix-construct/construct#141]
Riot stuck on first login since it tries to get e2ee stuff done, and chokes on many missing endpoints (or ones give unexpected replies).

Shall be fixed everything on Construct side (according to spec) then start complaining to Riot if the problem still persists.

(Reloading it makes it work just fine afterwards.)

**Comment by jevolk** [comment]
> chokes on many missing endpoints

Which endpoints?

> Ones give unexpected replies

What replies? Do you have more information from the browser devtools? Maybe a paste or screenshot of the errors?

> Reloading it makes it work just fine afterwards

Just clarifying, but a simple page refresh after the problem allows normal login to complete without endless spinner?

**Comment by jevolk** [comment]
> chokes on many missing endpoints

Which endpoints?

> Ones give unexpected replies

What replies? Do you have more information from the browser devtools? Maybe a paste or screenshot of the errors?

> Reloading it makes it work just fine afterwards

Just clarifying, but a simple page refresh after the problem allows normal login to complete without endless spinner?

**Comment by grinapo** [comment]
God only knows. My guesses would be these:
```
/_matrix/client/unstable/room_keys/version (404)
/_matrix/client/unstable/keys/device_signing/upload (401)
```

**Comment by grinapo** [comment]
> > Reloading it makes it work just fine afterwards
> 
> Just clarifying, but a simple page refresh after the problem allows normal login to complete without endless spinner?

Yes. So, you get a login page, you enter credentials and riot pulls up the Spinner of Eternity. 
Then you reload (carefully remove `/login` from the URL) and it'll end up in loading room data.
